clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -569,14 +569,32 @@ void PointerVariableConstraint::addArrayAnnotations(
   assert(CheckedArrs.empty());
 }
 
+
+bool PointerVariableConstraint::isTypedef(void) {
+  return IsTypedef;
+}
+
+void PointerVariableConstraint::setTypedef(TypedefNameDecl* T, std::string s) {
+  IsTypedef = true;
```

Comment:
add `assert` that these are non-null, since I assume that's what we are assuming?